agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -92,17 +103,22 @@ func (cs *CredentialSpecResource) Create() error {
 	var err error
 	var iamCredentials credentials.IAMRoleCredentials
 
+	domainlessGMSATask := false
 	executionCredentials, ok := cs.credentialsManager.GetTaskCredentials(cs.getExecutionCredentialsID())
 	if ok {
 		iamCredentials = executionCredentials.GetIAMRoleCredentials()
 	}
 
 	for credSpecStr := range cs.credentialSpecContainerMap {
-		credSpecSplit := strings.SplitAfterN(credSpecStr, "credentialspec:", 2)
+		credSpecSplit := strings.SplitAfterN(credSpecStr, ":", 2)
 		if len(credSpecSplit) != 2 {
 			seelog.Errorf("Invalid credentialspec: %s", credSpecStr)
 			continue
 		}
+		credSpecPrefix := credSpecSplit[0]
+		if credSpecPrefix == "credentialspecdomainless:" {
```

Comment:
Wait we are supporting csd in the old location too?

Reply:
We are supporting the following cases

the domain-joined cred in dockerSecurityOpts in task def 
the domain-joined cred in credentialSpecs in task def 
the domainless cred in credentialSpecs in task def 

If domain-joined appears in both dockerSecurityOpts and credentialSpecs, we prefer the value in credentialSpecs